### DISCUSSION TITLE
419 Page Expired

### DISCUSSION THREAD

**Discussion by dylanpatriarchi** [Laravel-Backpack/community-forum#901] [General] [answered]
Hello to everyone, i'm getting a problem with my csrf token, i tried to change APP_URL in .env file but it doesn't work. It's seems weird, actually i'm using Laravel 11 and Backpack 6.7

**Comment by dylanpatriarchi** [comment]
i found the solution by myself, for everyone who's struggling about this error: go to a file named datatables_logic(vendor/backpack/crud/src/resources/views/crud/inc, i suggest to publish this file) and add csrf token in the ajax call(row 264), after that it will work!!

**Comment by pxpm** [comment, marked as answer]
Hello @dylanpatriarchi 

Lucky us you decided to share the solution you found, because you should **NOT** be doing that. 👍 

Let me try to help you fix this. I am sure we (Backpack) should be adding the CSFR token on datatables ajax calls automatically for you.

The root cause for your error I think, it's because the assets are not properly loaded. The script that add the CSRF token automatically is located in `common.js` https://github.com/Laravel-Backpack/CRUD/blob/3ecaeffc603f0080ca4d5f7d108edca23e20b6af/src/resources/assets/js/common.js#L2

I would first run a `php artisan basset:check` to make sure basset is properly handling your files. 

In case it is, I would open the Network Tools and verify if the file I mentioned is beeing properly loaded. 

![image](https://github.com/Laravel-Backpack/community-forum/assets/7188159/2d217584-2a28-4a4a-a95b-683f4480b0a7)

If that doesn't help, please share more info about your setup, what's your `APP_URL` in your .env file, and what url you use to access in browser ? 

Cheers

**Reply by karandatwani92** [reply]
Yes, it seems related to `APP_URL`. @dylanpatriarchi, `APP_URL` must be the same as the URL you use to access the app.

**Reply by dylanpatriarchi** [reply]
Hi @pxpm @karandatwani92, my APP_URL is properly setted. The problem was Basset, thank you so much for the response now it's all working properly
